### DISCUSSION TITLE
Is there a way to make pyright 'recognize' django related name fields?

### DISCUSSION THREAD

**Discussion by tiredsapiens** [microsoft/pyright#9170] [General]
take for example a code snippet like this 
```
from django.db import models


class A(models.Model):
    id = models.IntegerField()


class B(models.Model):
    id = models.IntegerField()
    fk_A = models.ForeignKey(A, on_delete=models.CASCADE, related_name="fk_B")


a = A(id=1)
b = B(id=2, fk_A=a)
a.fk_B   #Here it says it cannot access attribute fk_B for class A

```
is there a way to make this work other than writing `#type ignore` every time i use a related name in django?(which i do alot)

**Comment by erictraut** [comment]
Django is using techniques to inject new attributes into a class without declaring them. In other words, it's using highly dynamic behaviors of the Python language that cannot be expressed using the [static type system](https://typing.readthedocs.io/en/latest/spec/index.html).

I don't know much about Django, but you may be able to add dummy class variable annotations, like this.

```python
class A(models.Model):
    id = models.IntegerField()
    fk_B: str
```

You might want to post your question to a forum with more Django users. They might have experience using Django with the Python static type system.

**Comment by slavugan** [comment]
Look at [django-types](https://github.com/sbdchd/django-types) it suggests such workaround:
```python
class Team(models.Model):
    if TYPE_CHECKING:
        user_set = models.manager.RelatedManager["User"]()
// OR
class Team(models.Model):
    user_set: models.Manager[User]
```
